### ISSUE TITLE
Application-Insights publishes for dependency flow

### ISSUE THREAD

**Issue by chcosta** [microsoft/ApplicationInsights-dotnet#1360]
Dependencies which are a part of source build, but do not publish using dependency flow end up causing incoherent builds. Source build is not able to determine which version of the asset is needed because it is not reported to BAR and then someone must manually go to a repo (like core-sdk) and trace back to find which version of the dependency is needed so that it can be built.

See https://github.com/dotnet/core-eng/issues/5360 for more info

**Comment by JohnTortugo** [comment]
Ping.

This might block completion of these issues: 
https://github.com/dotnet/core-eng/issues/5350
https://github.com/dotnet/core-eng/issues/5360

/cc @markwilkie

**Comment by TimothyMothra** [comment]
Hi @JohnTortugo I couldn't find your name in Outlook, can we talk offline?
I tried working on this last year but was blocked on several access issues and I ran out of time to complete this.
We're happy to comply but we can only do so much without support from your team. 

At this moment I don't remember the exact deliverables or how to validate successful completion.
If you have time to chat with me this week I think we can nail down some next steps.
